.circleci/config.yml

```diff
@@ -0,0 +1,136 @@
+      - run: ./scripts/time-cmd.sh ./gradlew --profile --parallel --stacktrace classes testClasses
+      - persist_to_workspace:
+          root: .
+          paths: [ . ]
+      - run: scripts/circle-ci/pr-changelog-status-check.sh
+
+  test:
+    parallelism: 8
+    docker:
+      - image: circleci/openjdk:8u171-jdk-node-browsers
+    resource_class: xlarge
+    environment:
+      GRADLE_OPTS: -Dorg.gradle.console=plain -Dorg.gradle.internal.launcher.welcomeMessageEnabled=false
+      _JAVA_OPTIONS: "-Xmx4g -XX:+HeapDumpOnOutOfMemoryError -verbose:gc -XX:+PrintGC -XX:+PrintGCDateStamps -XX:+PrintGCDetails -XX:-TraceClassUnloading -Xloggc:build-%t-%p.gc.log"
+      TERM: dumb
+      CIRCLE_TEST_REPORTS: /tmp/test-reports
+      CIRCLE_ARTIFACTS: /tmp/artifacts
+
+    steps:
+      - attach_workspace: { at: . }
+      - restore_cache:
+          key: gradle-wrapper-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}
+      - restore_cache:
+          key: gradle-cache-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}
+      - setup_remote_docker
+      - run: ./scripts/circle-ci/install-docker.sh
+      - run: sudo wget --retry-connrefused --waitretry=1 --read-timeout 20 --timeout 15 -t 10 -q -O /usr/local/bin/docker-compose https://github.com/docker/compose/releases/download/1.11.2/docker-compose-`uname -s`-`uname -m`
+      - run: sudo chmod +x /usr/local/bin/docker-compose
+      - run: docker info >/dev/null 2>&1 || sudo service docker start
+      - run:
+          command: ./scripts/circle-ci/pull-all-images.sh
+          background: true
+      - run:
+          command: ./scripts/circle-ci/track-memory-usage.sh
+          background: true
+      - run: ./scripts/circle-ci/download-moreutils.sh
+      - run:
+          command: ./scripts/time-cmd.sh ./scripts/circle-ci/run-circle-tests.sh
+          no_output_timeout: 600
+      - run: ./scripts/time-cmd.sh ./scripts/circle-ci/ensure-repo-clean.sh
+      - run: mkdir -p $CIRCLE_TEST_REPORTS/junit/
+      - run:
+          command: find . -type f -regex ".*/build/test-results/TEST-.*\.xml" -exec cp {} --parents $CIRCLE_TEST_REPORTS/junit/ \;
+          when: always
+      - store_test_results:
+          path: /tmp/test-reports/ # Circle doesn't do substitution in this path
+      - run: mkdir -p $CIRCLE_ARTIFACTS/checkstyle $CIRCLE_ARTIFACTS/findbugs
+      - run: if [ -d docs/build/html ]; then cp -r docs/build/html $CIRCLE_ARTIFACTS/docs; fi
+      - run: sudo apt-get install rsync
+      - run: find . -type d -regex ".*/build/reports/checkstyle" | (grep -v profile-client || true) | sed 's#\./\(\(.*\)/build/reports/checkstyle\)#rsync -uav \1/ $CIRCLE_ARTIFACTS/checkstyle/\2#' | bash
+      - run: find . -type d -regex ".*/build/reports/findbugs" | (grep -v profile-client || true) | sed 's#\./\(\(.*\)/build/reports/findbugs\)#rsync -uav \1/ $CIRCLE_ARTIFACTS/findbugs/\2#' | bash
+      - run: rsync -uav "build/reports/profile" "$CIRCLE_ARTIFACTS/"
```

Comment:
If you use `com.palantir.baseline-circleci` then you don't need this step.

https://github.com/palantir/gradle-baseline#circleci-plugin-compalantirbaseline-circleci